readthedocs/proxito/middleware.py

```diff
@@ -269,12 +270,22 @@ def add_hosting_integrations_headers(self, request, response):
         project_slug = getattr(request, "path_project_slug", "")
         version_slug = getattr(request, "path_version_slug", "")
 
-        if project_slug and version_slug:
-            addons = Version.objects.filter(
+        if project_slug:
+            force_addons = Project.objects.filter(
                 project__slug=project_slug,
-                slug=version_slug,
-                addons=True,
+                addons__isnull=False,
             ).exists()
+            if force_addons:
+                response["X-RTD-Force-Addons"] = "true"
+                return
```

Comment:
Why are we returning here? The logic of this function is a little confusing because we have `X-RTD-Force-Addons` and `X-RTD-Hosting-Integrations`, but I think they are just project and version level ways to enable addons?

Reply:
They are for different purposes:

* `X-RTD-Hosting-Integrations`: injects the new `readthedocs-addons.js`
* `X-RTD-Force-Addons`: injects the new `readthedocs-addons.js` and _remove all the files from the old integration_ (`readthedocs-doc-embed.js`, etc)

`X-RTD-Hosting-Integrations` is the one we are currently returning for projects using `build.commands`; and `X-RTD-Force-Addons` is going to be used when projects opt-in into the new beta addons -- even if they are not using `build.commands` and they have the old integrations already working.

Reply:
Gotcha.. probably good to define those headers somewhere -- perhaps they already are and we can just link that from the docstring?

Reply:
I added a docstring in this method.